src/main/java/graphql/normalized/ArgumentMaker.java

```diff
@@ -0,0 +1,110 @@
+package graphql.normalized;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import graphql.Internal;
+import graphql.execution.directives.QueryAppliedDirective;
+import graphql.execution.directives.QueryAppliedDirectiveArgument;
+import graphql.execution.directives.QueryDirectives;
+import graphql.language.Argument;
+import graphql.language.ArrayValue;
+import graphql.language.NullValue;
+import graphql.language.ObjectField;
+import graphql.language.ObjectValue;
+import graphql.language.Value;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.Map;
+
+import static graphql.collect.ImmutableKit.emptyMap;
+import static graphql.collect.ImmutableKit.map;
+import static graphql.language.Argument.newArgument;
+
+/**
+ * This class is a peer class and broken out of {@link ExecutableNormalizedOperationToAstCompiler} to deal with
+ * argument value making.
+ */
+@Internal
+class ArgumentMaker {
```

Comment:
as the comment says - breaking out code into its own class

Reply:
I like the class name, reminds me of this Monty Python sketch https://www.youtube.com/watch?v=uLlv_aZjHXc

> I'd like to have an argument please